e2e/cypress/integration/integrations/builtin_commands/user_status_commands_spec.js

```diff
@@ -0,0 +1,124 @@
+        cy.apiUpdateUserStatus('online');
+    });
+
+    testCases.forEach((testCase) => {
+        it(testCase.command, () => {
+            // # Post slash command
+            cy.postMessage(testCase.command + ' ');
+
+            verifyUserStatus(testCase);
+        });
+    });
+});
+
+describe('I18456 Built-in slash commands: user status via suggestion list', () => {
+    before(() => {
+        // # Login as user-1, go to "/" and set user status to online
+        cy.apiLogin('user-1');
+        cy.visit('/');
+        cy.apiUpdateUserStatus('online');
+    });
+
+    beforeEach(() => {
+        // # Type "/" on textbox
+        cy.get('#post_textbox').clear().type('/');
+    });
+
+    after(() => {
+        // # Set user status to offline
+        cy.apiUpdateUserStatus('online');
+    });
+
+    it('/away', () => {
+        const testCase = testCases[0];
+
+        // # Verify that the suggestion list is visible
+        cy.get('#suggestionList').should('be.visible').then((container) => {
+            // # Find command and click
+            cy.getByText(/\/away/, {container}).click({force: true});
+        });
+
+        cy.get('#post_textbox').type(' {enter}');
+
+        verifyUserStatus(testCase);
+    });
+
+    it('/dnd', () => {
+        const testCase = testCases[1];
+
+        // # Verify that the suggestion list is visible
+        cy.get('#suggestionList').should('be.visible').then((container) => {
+            // # Find command and click
+            cy.getByText(/\/dnd/, {container}).click({force: true});
+        });
+
+        // # Hit enter and verify user status
+        cy.get('#post_textbox').type(' {enter}');
+        verifyUserStatus(testCase);
+    });
+
+    it('/offline', () => {
+        const testCase = testCases[2];
+
+        // # Verify that the suggestion list is visible
+        cy.get('#suggestionList').should('be.visible').then((container) => {
+            // # Find command and click
+            cy.getByText(/\/offline/, {container}).click({force: true});
+        });
+
+        // # Hit enter and verify user status
+        cy.get('#post_textbox').type(' {enter}');
+        verifyUserStatus(testCase);
+    });
+
+    it('/online', () => {
+        const testCase = testCases[3];
+
+        // # Verify that the suggestion list is visible
+        cy.get('#suggestionList').should('be.visible').then((container) => {
+            // # Find command and click
+            cy.getByText(/\/online/, {container}).click({force: true});
+        });
+
+        // # Hit enter and verify user status
+        cy.get('#post_textbox').type(' {enter}');
+        verifyUserStatus(testCase);
+    });
+});
+
+function verifyUserStatus(testCase) {
+    // * Verify that the user status is as indicated
+    cy.get('#lhsHeader').find('svg').should('be.visible').and('have.attr', 'aria-label', testCase.ariaLabel);
+
+    // * Verify that ephemeral message is posted as expected
+    cy.getLastPostId().then((postId) => {
```

Comment:
You are expecting the last post to be made by the current user. But I don't see a step in this testcase where a post was made by the current user. Can you please include this step to make this testcase independent

Reply:
Hmm, not sure what you meant. The last post is expecting to be from `System` and not the current user.  This is a convenience function after user status change thru slash commands that generates corresponding ephemeral post.

Reply:
My Bad. Please ignore.